include/boost/circular_buffer/base.hpp

```diff
@@ -1475,7 +1476,8 @@ private empty_value<Alloc>
         if (full()) {
             if (empty())
                 return;
-            replace(m_last, value_type(::boost::forward<V>(value)));
+            destroy_item(m_last);
+            boost::allocator_construct(alloc(), boost::to_address(m_last), ::boost::forward<Args>(args)...);
```

Comment:
This has not been compiled with BOOST_NO_CXX11_VARIADIC_TEMPLATES defined because I do not think this would compile.

There is no variadic template parameter Args (args) here